puppet/modules/slave/manifests/postgresql.pp

```diff
@@ -16,6 +16,14 @@
           ensure => 'present',
           before => Class['postgresql::globals'],
         }
+        yumrepo { 'foreman-infra-el7':
+          baseurl =>  'https://yum.theforeman.org/infra/el7/',
+          enabled => true,
+        }
+        package { 'rh-postgresql12-postgresql-evr':
+          ensure  => 'present',
+          require => Package['postgresql::globals::server_package_name'],
```

Comment:
I don't think this literal package name exists. You can also use the wrapping class instead:
```suggestion
          require => Class['postgresql::server::install'],
```

 Also know that right now it's not guaranteed to be installed before the service starts. You may want to notify `postgresql::server::service` as well.

Reply:
I don't think restarting postgresql is required after installing an extension, but I can add it anyway to be safe

Reply:
I also don't know for sure if you need to, but as you say: it doesn't hurt.